src/api/catalog-search/catalog-search.ts

```diff
@@ -65,20 +65,25 @@ export class CatalogSearchAPI {
   public readonly constructFrameworks: CatalogConstructFrameworks;
 
   constructor(catalogData: CatalogPackage[], stats: PackageStats) {
-    const catalogMap = catalogData.reduce((map, pkg) => {
-      const { name, version } = pkg;
-      const id = [name, version].join("@");
-
-      const downloads = stats.packages[name]?.downloads?.npm ?? 0;
-
-      map.set(id, {
-        ...pkg,
-        downloads,
-        id,
-      });
-
-      return map;
-    }, new Map());
+    const catalogMap = catalogData
+      // Packages with the "construct-hub/hide-from-search" keyword are shadow-banned from search results
+      .filter(
+        (pkg) => !pkg.keywords?.includes("construct-hub/hide-from-search")
+      )
+      .reduce((map, pkg) => {
+        const { name, version } = pkg;
+        const id = [name, version].join("@");
+
+        const downloads = stats.packages[name]?.downloads?.npm ?? 0;
+
+        map.set(id, {
+          ...pkg,
+          downloads,
+          id,
+        });
+
+        return map;
+      }, new Map<string, any>());
```

Comment:
just two nits here:
- we could consolidate the `.filter` and `.reduce` by only calling `map.set` based on the filter condition. I dont' have any preference here
- instead of `new Map<string, any>()` perhaps we could use something like `new Map<string, ExtendedCatalogPackage>()`?

Reply:
> * we could consolidate the .filter and .reduce by only calling map.set based on the filter condition. I dont' have any preference here

Yeah I considered this, but I actually like the filter to be a separate stage in the pipeline. It makes it less likely to mis-interpret... The explaining comment feels a lot more local this way.

> instead of new Map<string, any>() perhaps we could use something like new Map<string, ExtendedCatalogPackage>()?

I had considered this. The previous syntax implicitly defaulted to `any` here, so I kept that.